### PR TITLE
Fix conversion of multiindexed pandas objects to sparse xarray objects

### LINKED ISSUE #4019: Sparse DataArray indexing gives incorrect results
Tested on xarray 0.15.1, sparse 0.9.1 and xarray 0.15.2.dev50+g3820fb77, sparse-0.9.1+26.ga9a6de2:
```
# Random sparse sample data
idx = pd.MultiIndex.from_product([np.arange(100), np.arange(100)], names=['a', 'b'])
s = pd.Series(np.random.RandomState(0).random(len(idx)), index=idx).sample(n=500, random_state=0)

da_dense = xr.DataArray.from_series(s, sparse=False)
da_sparse = xr.DataArray.from_series(s, sparse=True)

key = 23
print("Total:", da_dense.sum().values, da_sparse.sum().values)
print(f"loc[key]:", da_dense.loc[key].sum().values, da_sparse.loc[key].sum().values)
print(f"[key]:", da_dense[key].sum().values, da_sparse[key].sum().values)
print(f".isel(key):", da_dense.isel({'a': key}).sum().values, da_sparse.isel({'a': key}).sum().values)
print(f".sel(key):", da_dense.sel({'a': key}).sum().values, da_sparse.sel({'a': key}).sum().values)
```
Output:
```
Total: 253.0721848728631 253.07218487286306
loc[key]: 3.5885153944770103 0.0
[key]: 3.5885153944770103 0.0
.isel(key): 3.5885153944770103 0.0
.sel(key): 3.5885153944770103 0.0
```
It does appear that the underlying `sparse.COO` has the correct values:
```
np.nansum(da_dense.data[23])
3.5885153944770103

da_sparse.data.data[da_sparse.data.coords[0] == 23].sum()
3.5885153944770103
```
Happy to try to delve in deeper but if anyone knows off the top of their head what the issue might be that would be very welcome 🙂 

One other observation: the result isn't always 0, e.g.:
```
# key = 44
Total: 253.0721848728631 253.07218487286306
loc[key]: 2.868736626924726 1.1489982474345166
[key]: 2.868736626924726 1.1489982474345166
.isel(key): 2.868736626924726 1.1489982474345166
.sel(key): 2.868736626924726 1.1489982474345166
```

### PR DESCRIPTION
**Pull request by dcherian**
- [x] Closes #4019
 - [x] Tests added
 - [x] Passes `isort -rc . && black . && mypy . && flake8`
 - [x] Fully documented, including `whats-new.rst` for all changes and `api.rst` for new API

~Doesn't have a proper test. Need some help here~. cc @bnaul

### CODE REVIEW

**Comment by dcherian** [comment]
Ready for review. I added a test that checks if the `coords` for the underlying sparse object are the same when created in two ways
1. `from_series(..., sparse=False).data` converted to sparse using `sparse.COO.from_numpy`
2. `from_series(..., sparse=True).data`

File: xarray/core/dataset.py [hunk 1]
```diff
@@ -4537,7 +4537,7 @@ def _set_sparse_data_from_dataframe(
         idx = dataframe.index
         if isinstance(idx, pd.MultiIndex):
             coords = np.stack([np.asarray(code) for code in idx.codes], axis=0)
-            is_sorted = idx.is_lexsorted
+            is_sorted = idx.is_lexsorted()
```
Comment (shoyer) [inline xarray/core/dataset.py:4540]: Good catch! This is a little embarrassing :)

Comment (dcherian) [reply xarray/core/dataset.py:4540]: :) I blame Pandas API. `Index.is_sorted` and `Index.is_monotonic` are both properties! So expecting `is_lexsorted` to be a property is totally reasonable!

**Comment by dcherian** [comment]
Thanks for the reviews

**Pull request merged**